### PR TITLE
Improvement: Hypixel Mod API

### PR DESCRIPTION
**Pull request by appable0**
## What
Adds support for the Hypixel Mod API, using the hello and location packets to replace some functionality in HypixelData.

(I think the gradle build file and tweaker things have several issues)

## Changelog Improvements
+ Add some support for Hypixel Mod API. - appable
    * For location information, etc.

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/data/IslandType.kt [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
-    KUUDRA_ARENA("Kuudra"),
-    CRIMSON_ISLE("Crimson Isle"),
-    DWARVEN_MINES("Dwarven Mines"),
-    DUNGEON_HUB("Dungeon Hub"),
-    CATACOMBS("Catacombs"),
+    THE_END("The End", "combat_3"),
+    KUUDRA_ARENA("Kuudra", "instanced"),
+    CRIMSON_ISLE("Crimson Isle", "crimson_isle"),
+    DWARVEN_MINES("Dwarven Mines", "mining_3"),
+    DUNGEON_HUB("Dungeon Hub", "dungeon)hub"),
```
Comment (jani270) [inline src/main/java/at/hannibal2/skyhanni/data/IslandType.kt]:
```suggestion
    DUNGEON_HUB("Dungeon Hub", "dungeon_hub"),
```

**Comment by hannibal002** [comment]
Please keep the existing logic as a backup system. We don't want to be at the mercy of Hypixel not disabling/changing the mod API.

**Comment by appable0** [comment]
Closing as I haven't had free time (:c) and it seems like Empa is working on it now!

**Comment by hannibal002** [comment]
Totally forgot about this PR, so sorry :(

**Comment by ItsEmpa** [comment]
i didnt know this pr existed 😭

**Pull request closed without merging**